packages/compass-components/src/hooks/use-navigation-tree.ts

```diff
@@ -0,0 +1,246 @@
+}): React.HTMLAttributes<HTMLUListElement> & {
+  ref: React.RefObject<HTMLUListElement>;
+} {
+  const ref = useRef<HTMLUListElement>(null);
+
+  const onKeyDown = useCallback(
+    (evt: React.KeyboardEvent<HTMLUListElement>) => {
+      if (!ref.current) {
+        return;
+      }
+
+      const nodes = findAllTreeItems(ref.current);
+      const currentFocusElement = nodes.find(
+        (node) => node === document.activeElement
+      );
+
+      if (!currentFocusElement) {
+        return;
+      }
+
+      const currentFocusElementIndex = nodes.indexOf(currentFocusElement);
+
+      let nextFocusElementIndex = -1;
+
+      if (evt.key === 'Home') {
+        nextFocusElementIndex = 0;
+      }
+
+      if (evt.key === 'End') {
+        nextFocusElementIndex = nodes.length - 1;
+      }
+
+      if (evt.key === 'ArrowDown') {
+        evt.preventDefault();
+        evt.stopPropagation();
+
+        nextFocusElementIndex = currentFocusElementIndex + 1;
+      }
+
+      if (evt.key === 'ArrowUp') {
+        evt.preventDefault();
+        evt.stopPropagation();
+
+        nextFocusElementIndex = currentFocusElementIndex - 1;
+      }
+
+      if (evt.key === 'ArrowRight') {
+        evt.stopPropagation();
+
+        if (currentFocusElement.getAttribute('aria-expanded') === 'false') {
+          onExpandedChange(currentFocusElement.dataset.id as string, true);
+        }
+
+        if (currentFocusElement.getAttribute('aria-expanded') === 'true') {
+          const currentFocusElementLevel = Number(
+            currentFocusElement.getAttribute('aria-level')
+          );
+          const maybeNextFocusElementIndex = currentFocusElementIndex + 1;
+          const maybeNextFocusElement = nodes[maybeNextFocusElementIndex];
+          if (
+            maybeNextFocusElement &&
+            Number(maybeNextFocusElement.getAttribute('aria-level')) ===
+              currentFocusElementLevel + 1
+          ) {
+            nextFocusElementIndex = maybeNextFocusElementIndex;
+          }
+        }
+      }
+
+      if (evt.key === 'ArrowLeft') {
+        evt.stopPropagation();
+
+        if (currentFocusElement.getAttribute('aria-expanded') === 'true') {
+          onExpandedChange(currentFocusElement.dataset.id as string, false);
+        } else {
+          const parentNode = findParentTreeItem(currentFocusElement);
+          if (parentNode) {
+            nextFocusElementIndex = nodes.indexOf(parentNode);
+          }
+        }
+      }
+
+      if (evt.key === '*') {
+        evt.stopPropagation();
+
+        for (const node of findAllNodesInSet(currentFocusElement)) {
+          if (node.getAttribute('aria-expanded') === 'false') {
+            onExpandedChange(node.dataset.id as string, true);
+          }
+        }
+      }
+
+      if (/^\p{Letter}$/u.test(evt.key)) {
+        evt.stopPropagation();
+
+        const letter = evt.key.toLocaleLowerCase();
+        const maybeNode = nodes
+          .slice(currentFocusElementIndex + 1)
```

Comment:
Good catch! Missed this part of the spec, will fix.

Yeah, it's basically catching all the unicode symbols that have LETTER property. This `\p` is called [unicode property escape](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Unicode_Property_Escapes) and it allows you to match by unicode properties. There is a ton of cases where it can simplify targeting your regex, for example you can catch all emojis with `{Emoji_Presentation}` group